pkg/systemd/services/service.jsx

```diff
@@ -21,50 +21,132 @@ import React from "react";
 import { Breadcrumb, BreadcrumbItem } from "@patternfly/react-core/dist/esm/components/Breadcrumb/index.js";
 import { Page, PageSection } from "@patternfly/react-core/dist/esm/components/Page/index.js";
 import { Gallery, GalleryItem } from "@patternfly/react-core/dist/esm/layouts/Gallery/index.js";
+import { ExclamationCircleIcon } from '@patternfly/react-icons';
 
+import { EmptyStatePanel } from "cockpit-components-empty-state.jsx";
 import { ServiceDetails } from "./service-details.jsx";
 import { LogsPanel } from "cockpit-components-logs-panel.jsx";
 import { superuser } from 'superuser';
 import { WithDialogs } from "dialogs.jsx";
 
+import s_bus from "./busnames.js";
+
 import cockpit from "cockpit";
 
 const _ = cockpit.gettext;
 
+function debug() {
+    if (window.debugging == "all" || window.debugging?.includes("service-details"))
+        console.debug.apply(console, arguments);
+}
+
 export class Service extends React.Component {
     constructor(props) {
         super(props);
 
         this.state = {
-            /* The initial load of the Services page will not call GetAll for units Properties
-             * since ListUnits API call already has provided us with a subset of the Properties.
-             * As a result, properties like the 'Requires' are not present in the state at this point.
-             * If it's the first time to open this service's details page we need to fetch
-             * the unit properties by calling getUnitByPath.
-             */
-            shouldFetchProps: props.unit.Names === undefined,
+            unit_id: null,
+            unit_props: null,
+            error: null,
+            reloading: false,
         };
+
+        this.path = null;
+        this.subscriptions = [];
+    }
+
+    async componentDidMount() {
+        const dbus = this.props.dbusClient;
+        try {
+            const [path] = await dbus.call(s_bus.O_MANAGER, s_bus.I_MANAGER, "LoadUnit", [this.props.unitId]);
+            this.path = path;
+        } catch (ex) {
+            this.setState({ error: ex.toString() });
+            return;
+        }
```

Comment:
These 4 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18422-20230307-104236-77724996-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)